lambda-layer/build-layer.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+SOURCEDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+
+## revert https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7970
+OTEL_VERSION="1.32.1"
+ADOT_VERSION="1.32.1"
+
+
+## Clone and Patch the OpenTelemetry Java Instrumentation Repository
+git clone https://github.com/open-telemetry/opentelemetry-java-instrumentation.git
+pushd opentelemetry-java-instrumentation
+git checkout v${OTEL_VERSION} -b tag-v${OTEL_VERSION}
+
+# This patch is for Lambda related context propagation
+patch -p1 < "$SOURCEDIR"/patches/opentelemetry-java-instrumentation.patch
+
+# There is another patch in the .github/patches directory for other changes. We should apply them too for consistency.
+patch -p1 < "$SOURCEDIR"/../.github/patches/opentelemetry-java-instrumentation.patch
+
+git add -A
+git commit -m "Create patch version"
+./gradlew publishToMavenLocal
+popd
+rm -rf opentelemetry-java-instrumentation
+
+
+## Build the ADOT Java from current source
+pushd "$SOURCEDIR"/..
+patch  -p1 < "${SOURCEDIR}"/patches/aws-otel-java-instrumentation.patch
+CI=false ./gradlew publishToMavenLocal -Prelease.version=${ADOT_VERSION}-adot-lambda1
+popd
+
+
+## Build ADOT Lambda Java SDK Layer Code
+./gradlew build
+
+
+## Copy ADOT Java Agent downloaded using Gradle task and bundle it with the Lambda handler script
+cp "$SOURCEDIR"/build/javaagent/aws-opentelemetry-agent*.jar ./opentelemetry-javaagent.jar
+zip -qr opentelemetry-javaagent-layer.zip opentelemetry-javaagent.jar otel-handler
```

Comment:
Is the LAMBDA_EXEC_WRAPPER script added in zip?

Reply:
Yes. It's the [`otel-handler`](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/953/files#diff-3f19c5849b10c7d9b4d2a3e4bee2cf89de66fc1cefe5fca727b114c0df3fa5ff) in this line.

Reply:
rename to otel-instrument, to align with other language?

Reply:
I would prefer keeping it `otel-handler` to align with [the upstream naming convention](https://github.com/open-telemetry/opentelemetry-lambda/blob/main/java/layer-javaagent/scripts/otel-handler), since the script is pretty much based off of it. This way AWS devs can easily find references in upstream.

BTW, `otel-instrument` used by Python [aligns with upstream](https://github.com/open-telemetry/opentelemetry-lambda/blob/main/python/src/otel/otel_sdk/otel-instrument), but I think for Node we should have called it `otel-handler` since that's what [the upstream does](https://github.com/open-telemetry/opentelemetry-lambda/blob/main/nodejs/packages/layer/scripts/otel-handler).